### PR TITLE
gtk-vnc: 0.9.0 → 1.0.0

### PR DESCRIPTION
**Pull request by jtojnar**
https://mail.gnome.org/archives/gtk-vnc-list/2019-August/msg00000.html

Meson port 🎉

### CODE REVIEW

**Comment by jtojnar** [comment]
:heavy_check_mark: Successfully built all dependent packages:

```
nix-build -A gnome3.gnome-boxes -A gnome3.vinagre -A virt-viewer -A virtmanager
```

**Comment by 7c6f434c** [comment]
Why is `gvncviewer` binary no longer built?

**Comment by jtojnar** [comment]
It is built but no longer installed. Since it is just an example it is probably not needed to be installed.

**Comment by 7c6f434c** [comment]
Wait, what does it lack? Because I used it as a viewer from time to time.

**Comment by jtojnar** [comment]
I am not sure what it lacks but it is not a production-ready application – it is an [example](https://gitlab.gnome.org/GNOME/gtk-vnc/blob/94377d8716d3b4c7e78d763ac074fa937415fb44/examples/gvncviewer.c) of how to use the library. So it makes sense that during the build system port, they [stopped installing it](https://gitlab.gnome.org/GNOME/gtk-vnc/blob/94377d8716d3b4c7e78d763ac074fa937415fb44/examples/meson.build#L12).

**Comment by mmahut** [comment]
That's fair, if they do to maintain it and there is an alternative we should probably not ship it as a binary.

File: pkgs/tools/admin/gtk-vnc/default.nix [hunk 1]
```diff
@@ -1,29 +1,59 @@
...
+    pkgconfig
+    gobject-introspection
+    vala
+    gettext
+    perl # for pod2man
   ];
 
-  configureFlags = [
-    "--with-examples"
+  buildInputs = [
```
Comment (worldofpeace) [inline pkgs/tools/admin/gtk-vnc/default.nix]: [gdk-pixbuf](https://gitlab.gnome.org/GNOME/gtk-vnc/blob/v1.0.0/meson.build#L80), [zlib](https://gitlab.gnome.org/GNOME/gtk-vnc/blob/v1.0.0/meson.build#L90)?

File: pkgs/tools/admin/gtk-vnc/default.nix [hunk 2]
```diff
@@ -1,29 +1,59 @@
...
+    gnutls
+    cairo
+    glib
+    libgcrypt
+    cyrus_sasl
+    libpulseaudio
+    gtk3
   ];
 
   passthru = {
```
Comment (worldofpeace) [inline pkgs/tools/admin/gtk-vnc/default.nix]: `pname` for `packageName`?

**Review by worldofpeace — CHANGES_REQUESTED** [review]

File: pkgs/tools/admin/gtk-vnc/default.nix [hunk 3]
```diff
@@ -1,29 +1,59 @@
...
-    url = "mirror://gnome/sources/gtk-vnc/${stdenv.lib.versions.majorMinor version}/${name}.tar.xz";
-    sha256 = "1dya1wc9vis8h0fv625pii1n70cckf1xjg1m2hndz989d118i6is";
+    url = "mirror://gnome/sources/${pname}/${stdenv.lib.versions.majorMinor version}/${pname}-${version}.tar.xz";
+    sha256 = "1060ws037v556rx1qhfrcg02859rscksrzr8fq11himdg4d1y6m8";
   };
 
-  nativeBuildInputs = [
-    python3 pkgconfig intltool libtool gobject-introspection vala
+  patches = [
+    (fetchpatch {
```
[the remark this answers is not in the document]
Comment (jtojnar) [reply pkgs/tools/admin/gtk-vnc/default.nix]: Yay, I always ask other people for that but keep forgetting it myself :stuck_out_tongue:

[the remark this answers is not in the document]
Comment (worldofpeace) [reply pkgs/tools/admin/gtk-vnc/default.nix]: :rofl: The tables have turned

**Pull request merged**